src/Pagination/Pagination.tsx

```diff
@@ -23,37 +24,59 @@ export interface PaginationProps {
 }
 
 export const Pagination: React.FunctionComponent<PaginationProps> = React.memo((props: PaginationProps): React.ReactElement<void> => {
-    const Initialoffset: number = props.offset ? props.offset : 10;
+
+    const [list, setList] = React.useState<Array<number>>([]);
+    const [dotnavList, setDotnavList] = React.useState<Array<number>>([]);
+    const [pagingSize, setPagingSize] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        const initialOffset: number = props.offset ? props.offset : 10;
+        setPagingSize(Math.ceil(props.size / initialOffset));
+    }, [props.offset, props.pagingLength, props.size]);
+
+    React.useEffect(() => {
+        const intialLength: number = props.pagingLength ? props.pagingLength : 5;
+        generateList(props.value, intialLength);
+    }, [pagingSize, props.value]);
```

Comment:
maybe props.pagingLength should be added here.

Reply:
```suggestion
    }, [pagingSize, props.value, props.pagingLength]);
```